addon/mixins/application-route-mixin.js

```diff
@@ -54,23 +55,25 @@ export default Ember.Mixin.create({
     @private
   */
   beforeModel(transition) {
-    this._super(transition);
-    if (!this.get('_authEventListenersAssigned')) {
-      this.set('_authEventListenersAssigned', true);
-      Ember.A([
-        'sessionAuthenticationSucceeded',
-        'sessionAuthenticationFailed',
-        'sessionInvalidationSucceeded',
-        'sessionInvalidationFailed',
-        'authorizationFailed'
-      ]).forEach((event) => {
-        this.get(Configuration.base.sessionPropertyName).on(event, Ember.run.bind(this, function() {
-          Array.prototype.unshift.call(arguments, event);
-          let target = routeEntryComplete ? this : transition;
-          target.send.apply(target, arguments);
-        }));
-      });
-    }
+    setup(this.container).finally(() => {
+      this._super(transition);
+      if (!this.get('_authEventListenersAssigned')) {
```

Comment:
Good point, no need to check if listeners are assigned as `ApplicationRoute.beforeModel` is only called once per page load.